projects/plugins/jetpack/extensions/blocks/premium-content/_inc/access-check.php

```diff
@@ -96,7 +96,22 @@ function current_visitor_can_access( $attributes, $block ) {
 
 	$paywall      = subscription_service();
 	$access_level = Token_Subscription_Service::POST_ACCESS_LEVEL_PAID_SUBSCRIBERS; // Only paid subscribers should be granted access to the premium content
-	$can_view     = $paywall->visitor_can_view_content( array( $selected_plan_id ), $access_level );
+	$tier_ids     = \Jetpack_Memberships::get_all_newsletter_plan_ids();
+	if ( in_array( $selected_plan_id, $tier_ids, true ) ) {
```

Comment:
would it make sense to do it inside `visitor_can_view_content`?  Could be reused in other places

Reply:
`visitor_can_view_content` is also used in other places, but we might not want it (it would run as a double for Newsletter).

It might be a good part for a good refactor, but I don't feel like tackling that right now...